databases/week3/meal_sharing.sql

```diff
@@ -0,0 +1,106 @@
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)VALUES("It was ok","it takes too much time",3,2.5,"2021-10-4");
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)VALUES("not happy","I had better",4,1,"2021-10-5");
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)VALUES("super","Definitly make again",5,5,"2021-10-6");
+--Create these queries 
+SELECT * from meal;
+INSERT INTO meal (`title`,`description`,`location`,`when`,`max_reservations`,`price`,`created_date`)VALUES("calzone","how to make spinach calzone","online","2021-10-01","4","400","2021-9-1");
+SELECT * from meal WHERE id = 3;
+--UPDATE table_name
+--SET column1 = value1, column2 = value2, ...
+--WHERE condition;
+UPDATE  meal 
+SET description = "it cant be more Italian"
+WHERE id = 6;
+DELETE from meal WHERE id = 2;
+--Reservation queries
+SELECT * from reservation;
+INSERT INTO reservation(`number_of_guests`,`meal_id`,`created_date`,`contact_phonenumber`,`contact_name`,`contact_email`)VALUES("4","7","2021-9-2","44455566","Sunil","sunilkumarmahajan@gmail.com");
+SELECT * from reservation WHERE id = 4;
+UPDATE reservation
+SET created_date = '2021-10-4'
+WHERE id = 4;
+DELETE from reservation WHERE id = 5;
+--review data 
+SELECT * from review;
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)VALUES("super","easy to cook",7,4,"2021-10-9");
+select * from review WHERE id =7;
+UPDATE review
+set title = "super good", stars = "4.5"
+WHERE id = 7;
+DELETE from review WHERE id = 6;
+--Additional queries
+--Get meals that has a price smaller than a specific price fx 90
+SELECT * from meal where price < 400;
+--Get meals that still has available reservations
+INSERT INTO reservation(`number_of_guests`,`meal_id`,`created_date`,`contact_phonenumber`,`contact_name`,`contact_email`)VALUES('4',"7","2021-9-2",null,null,null);
+DELETE from reservation where id=7; 
+INSERT INTO meal (`title`,`description`,`location`,`when`,`max_reservations`,`price`,`created_date`)VALUES("pasta","how to make great olive pasta","online","2021-10-011",NULL,"350","2021-9-1");
+SELECT * from meal WHERE max_reservations = NULL;
+--Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT * from meal WHERE title LIKE '%pasta%';
+--Get meals that has been created between two dates
+UPDATE meal
+SET created_date = '2021-09-3'
+WHERE id = 3;
+SELECT created_date
+FROM meal 
+WHERE 2021-09-01 OR 2021-09-03;
+--Get only specific number of meals fx return only 5 meals
+SELECT * from meal WHERE id > 3;
+--Get the meals that have good reviews
+SELECT * from review WHERE stars > 4;
+--Get reservations for a specific meal sorted by created_date
+SELECT meal.id, reservation.meal_id, meal.title, 
+reservation.number_of_guests, reservation.created_date FROM meal, reservation
+WHERE meal.title LIKE "carbonerra" and meal.id=reservation.meal_id
```

Comment:
Why we need to select this specific meal(WHERE meal. title LIKE "carbonara")? You don't need it. You have to sort out all meals according to their create_date
hint
```
SELECT meal.Title, reservation.meal_Id, SUM(reservation.Number_of_guests), reservation.Created_date AS Res_created_date
FROM reservation
   JOIN meal ON (meal.id=reservation.meal_id)
GROUP BY reservation.created_date,meal.Title
ORDER BY reservation.created_date;
```

Reply:
it gives me back this error "  SELECT list is not in GROUP BY clause and contains nonaggregated column 'meal_sharing.reservation.meal_id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by".?!!

Reply:
Should give out like this
![image](https://user-images.githubusercontent.com/61273043/137023717-ed5c000d-2992-4db3-85c2-1d94b128508d.png)